src/ui/public/filter_bar/editable_filter.js

```diff
@@ -0,0 +1,33 @@
+define(function (require) {
+  var _ = require('lodash');
+  var angular = require('angular');
+
+  require('ui/modules')
+  .get('kibana')
+  .directive('editableFilter', function () {
+    return {
+      restrict: 'A',
+      require: 'ngModel',
+      link: function ($scope, $el, attrs, ngModelCntrl) {
+        ngModelCntrl.$formatters.push(toJSON);
+        ngModelCntrl.$parsers.push(fromJSON);
+
+        function fromJSON(value) {
+          try {
+            value = JSON.parse(value);
+            var definedFilter = _.keys(value).length > 0;
```

Comment:
Can you move this to an attribute option? This would allow us to reuse this validator elsewhere in the app, eg `<textarea ng-model="myObject" require-keys=false>`